src/commands/readme.rs

```diff
@@ -0,0 +1,238 @@
+                Self::UpdateNeeded => "Updated needed",
+                Self::UpToDate => "Up-to-date",
+            }
+        )
+    }
+}
+
+pub fn check_pkg_readme<'a>(
+    ws: &Workspace<'a>,
+    pkg_path: &Path,
+    pkg_manifest: &Manifest,
+) -> Result<(), String> {
+    let c = ws.config();
+
+    let mut pkg_source = find_entrypoint(pkg_path, pkg_manifest)?;
+    let readme_path = pkg_path.join("README.md");
+
+    c.shell()
+        .status("Checking", format!("Readme for {}", &pkg_manifest.name()))
+        .map_err(|e| format!("{:}", e))?;
+
+    let pkg_readme = fs::read_to_string(readme_path.clone());
+    match pkg_readme {
+        Ok(pkg_readme) => {
+            // Try to find readme template
+            let template_path = find_readme_template(&ws.root(), &pkg_path)?;
+
+            let new_readme = generate_readme(&pkg_path, &mut pkg_source, template_path)?;
+            if Sha1::from(pkg_readme) == Sha1::from(new_readme) {
+                Ok(())
+            } else {
+                Err(CheckReadmeResult::UpdateNeeded.to_string())
+            }
+        }
+        Err(_err) => Err(CheckReadmeResult::Missing.to_string()),
+    }
+}
+
+pub fn gen_all_readme<'a>(
+    packages: &Vec<Package>,
+    ws: &Workspace<'a>,
+    readme_mode: GenerateReadmeMode,
+) -> Result<(), Box<dyn Error>> {
+    let c = ws.config();
+
+    c.shell().status("Generating", "Readme files")?;
+    for pkg in packages.iter() {
+        let pkg_path = pkg.manifest_path().parent().expect("Folder exists");
+        gen_pkg_readme(ws, &pkg_path, &pkg.manifest(), &readme_mode)
+            .map_err(|e| format!("Failure generating Readme for {:}: {}", pkg.name(), e))?
+    }
+
+    Ok(())
+}
+
+pub fn gen_pkg_readme<'a>(
+    ws: &Workspace<'a>,
+    pkg_path: &Path,
+    pkg_manifest: &Manifest,
+    mode: &GenerateReadmeMode,
+) -> Result<(), String> {
+    let c = ws.config();
+    let root_path = ws.root();
+
+    let mut pkg_source = find_entrypoint(pkg_path, pkg_manifest)?;
+    let readme_path = pkg_path.join("README.md");
+
+    let pkg_readme = fs::read_to_string(readme_path.clone());
+    match (mode, pkg_readme) {
+        (GenerateReadmeMode::IfMissing, Ok(_existing_readme)) => {
+            c.shell()
+                .status(
+                    "Skipping",
+                    format!("{}: Readme already exists.", &pkg_manifest.name()),
+                )
+                .map_err(|e| format!("{:}", e))?;
+
+            Ok(())
+        }
+        (mode, existing_res) => {
+            let template_path = find_readme_template(&ws.root(), &pkg_path)?;
+            c.shell()
+                .status(
+                    "Generating",
+                    format!(
+                        "Readme for {} (template: {:?})",
+                        &pkg_manifest.name(),
+                        match &template_path {
+                            Some(p) => p.strip_prefix(&root_path).unwrap().to_str().unwrap(),
```

Comment:
can we be sure this never fails? why not use `?`?
```suggestion
                            Some(p) => p.strip_prefix(&root_path)?.to_str()?,
```